library/general/src/lib/yast2/log_dir_rotator.rb

```diff
@@ -0,0 +1,195 @@
+  #   mv #{NAME}    #{NAME}-01
+  #
+  # System error are logged and no exception is raised by the functions.
+  #
+  # TODO This class is based on dump_manager.rb and could be used there.
+  class LogDirRotator
+    include Yast::Logger
+
+    # The number of old log directories to keep. This is in addition
+    # to the current one. Child classes can override the value. Max
+    # value is 99.
+    KEEP_OLD_LOG_DIRS = 3
+
+    # The name of the directory in /var/log/YaST2 to create, rotate
+    # and delete. Child classes should override the value.
+    NAME = "undefined".freeze
+
+    # Prepare (create, rotate and delete) the log directory.
+    def prepare
+      return if Yast::Mode.test
+      begin
+        log.info "preparing log dir #{self.class::NAME}"
+        rotate_log_dirs
+        FileUtils.mkdir_p(log_dir)
+      rescue SystemCallError => e
+        log.error e.to_s
+      end
+    end
+
+    # Copy a file to the log directory.
+    #
+    # @param src [String] name of source
+    # @param dest [String] basename of destination
+    def copy(src, dest)
+      return if Yast::Mode.test
+      begin
+        FileUtils.cp(src, log_dir + "/" + dest)
+      rescue SystemCallError => e
+        log.error e.to_s
+      end
+    end
+
+    # Write a file in the log directory.
+    #
+    # @param dest [String] basename of destination
+    # @param content [String] content to write
+    def write(dest, content)
+      return if Yast::Mode.test
+      begin
+        File.open(log_dir + "/" + dest, "w") { |file| file.write(content) }
+      rescue SystemCallError => e
+        log.error e.to_s
+      end
+    end
+
+    # Return a suitable name for the devicegraph log directory
+    # depending on the YaST mode (installation / installed system).
+    #
+    # @return [String] directory name with full path
+    def log_dir
+      dir = installation? ? "#{self.class::NAME}-inst" : self.class::NAME
+      base_dir + "/" + dir
+    end
+
+  private
+
+    # Return true if this is some installation mode: installation, update
+    #
+    # @return [Boolean]
+    def installation?
+      Yast::Mode.installation || Yast::Mode.update
+    end
+
+    # Rotate the log directories, depending on current YaST mode:
+    #
+    # During installation (or update or AutoYaST), clear and remove any old
+    # /var/log/YaST2/#{NAME}-inst directory.
+    #
+    # In the installed system, keep a number of old log directories, remove
+    # any older ones in /var/log/YaST2, and rename the ones to keep:
+    #
+    #   rm -rf #{NAME}-03
+    #   mv #{NAME}-02 #{NAME}-03
+    #   mv #{NAME}-01 #{NAME}-02
+    #   mv #{NAME}    #{NAME}-01
+    #
+    # This will NOT create any new log directory.
+    def rotate_log_dirs
+      return unless File.exist?(base_dir)
+      if installation?
+        kill_old_log_dirs([File.basename(log_dir)])
+      else
+        log_dirs = old_log_dirs.sort
+        keep_dirs = log_dirs.shift(self.class::KEEP_OLD_LOG_DIRS)
```

Comment:
this won;t work when KEEP_OLD_LOG_DIRS will be bigger then 10 as it is string sorting. See this output of irb:
```
irb(main):004:0> a = []; 11.times { |i| a << "t#{i}" }; a.sort
=> ["t0", "t1", "t10", "t2", "t3", "t4", "t5", "t6", "t7", "t8", "t9"]
```

Reply:
The dirs are t-01, t-02, ... so it does work.

Reply:
ah, right